pkg/apis/pipeline/v1/task_validation.go

```diff
@@ -305,13 +309,25 @@ func validateStep(ctx context.Context, s Step, names sets.String) (errs *apis.Fi
 				Paths:   []string{"volumeMounts"},
 			})
 		}
+		if len(s.Results) > 0 {
+			errs = errs.Also(&apis.FieldError{
+				Message: "results cannot be used with Ref",
+				Paths:   []string{"results"},
+			})
+		}
+
 	} else {
 		if len(s.Params) > 0 {
 			errs = errs.Also(&apis.FieldError{
 				Message: "params cannot be used without Ref",
 				Paths:   []string{"params"},
 			})
 		}
+		if len(s.Results) > 0 {
+			if !config.FromContextOrDefaults(ctx).FeatureFlags.EnableStepActions {
+				return apis.ErrGeneric("feature flag %s should be set to true in order to use Results in Steps.", config.EnableStepActions)
```

Comment:
So the original field can be non-empty. 
i.e. if Users choose to use results for inlined steps.

They just need to enable-step-actions for this, no?

Reply:
done. I made use of `IsInCreate` and `IsInUpdate` for this.

Reply:
The controller context has these both set to `false`. But for web hook events, these are set appropriately. This means that during web hook validation, we can enforce this. In the controller logic, this does not really need to be validated since it made it past the web hook.

Reply:
Mostly sgtm, but `apis.IsInUpdate(ctx)` isn't sufficient, since it would still reject updates to unrelated fields / finalizers if the flag was disabled. :(

We also need to check if the field we want to check has changed - you can grab the old object via [GetBaseline](https://pkg.go.dev/knative.dev/pkg@v0.0.0-20231120182734-703c8b0d5c34/apis#GetBaseline)